cmd/xl-storage-format-v2.go

```diff
@@ -1190,7 +1190,7 @@ func (x *xlMetaV2) DeleteVersion(fi FileInfo) (string, bool, error) {
 			}
 			return "", len(x.versions) == 0, err
 		case ObjectType:
-			if updateVersion {
+			if updateVersion && !fi.Deleted {
```

Comment:
fi.Deleted will never be true for ObjectType, only for delete markers - this additional if condition can be avoided

Reply:
@poornas this is the main bug fix in this PR - the reason is `ObjectType` is the list of current versions on the disk - when you want to create a `null` version Marker you don't want to be updating the version on the disk - that is the reason why `DeleteMarker` was never created on the disk. 

This PR addresses that because `updateVersion` is true since fi.MarkDeleted is true which leads down this path so a client sees that DeleteObject() is successful however nothing was purged on the disk.